client/blocks/inline-help/inline-help-search-results.jsx

```diff
@@ -0,0 +1,186 @@
+				title: this.props.translate( 'Uploading custom plugins and themes' ),
+				description: this.props.translate(
+					'Learn more about installing a custom theme or plugin using the Business plan.'
+				),
+			},
+			{
+				link: 'https://en.support.wordpress.com/all-about-domains/',
+				title: this.props.translate( 'All About Domains' ),
+				description: this.props.translate(
+					'Set up your domain whether it’s registered with WordPress.com or elsewhere.'
+				),
+			},
+			{
+				link: 'https://en.support.wordpress.com/start/',
+				title: this.props.translate( 'Get Started' ),
+				description: this.props.translate(
+					'No matter what kind of site you want to build, our five-step checklists will get you set up and ready to publish.'
+				),
+			},
+			{
+				link: 'https://en.support.wordpress.com/settings/privacy-settings/',
+				title: this.props.translate( 'Privacy Settings' ),
+				description: this.props.translate(
+					'Limit your site’s visibility or make it completely private.'
+				),
+			},
+		];
+		return helpfulResults;
+	}
+
+	renderSearchResults() {
+		if ( isEmpty( this.props.searchQuery ) ) {
+			// not searching
+			return this.renderContextHelp();
+		}
+
+		if ( this.props.isSearching ) {
+			// search, but no results so far
+			return (
+				<PlaceholderLines />
+			);
+		}
+
+		if ( isEmpty( this.props.searchResults ) ) {
+			// search done, but nothing found
+			return (
+				<div>
+					<p className="inline-help__empty-results">No results.</p>
+					{ this.renderContextHelp() }
+				</div>
+			);
+		}
+
+		// found something
+		const links = this.props.searchResults;
+		return (
+			<ul className="inline-help__results-list">{ links && links.map( this.renderHelpLink ) }</ul>
+		);
+	}
+
+	renderContextHelp() {
+		const links = this.getContextResults();
+		return (
+			<ul className="inline-help__results-list">
+				{ links && links.map( this.renderHelpLink ) }
+			</ul>
+		);
+	}
+
+	getSelectedUrl = () => {
+		if ( this.props.selectedResult === -1 ) {
+			return false;
+		}
+
+		const links = this.props.searchResults || this.getContextResults();
+		const selectedLink = links[ this.props.selectedResult ];
+		if ( ! selectedLink || ! selectedLink.link ) {
+			return false;
+		}
+		return selectedLink.link;
+	}
+
+	componentWillUpdate( nextProps ) {
+		if ( nextProps.shouldOpenSelectedResult ) {
+			const url = this.getSelectedUrl();
+			this.followHelpLink( url );
+			this.props.didOpenResult();
+			window.location = url;
+			return;
+		}
+	}
+
+	followHelpLink = ( url ) => {
+		const payload = {
+			searchQuery: this.props.searchQuery,
+			currentUrl: window.location.href,
+			resultUrl: url,
+		};
+		return () => {
+			tracks.recordEvent( 'calypso_inline-help_follow-link', payload );
+		};
+	}
+
+	renderHelpLink = ( link, index ) => {
+		const classes = [
+			'inline-help__results-item',
+			this.props.selectedResult === index && 'is-selected',
+		].filter( Boolean );
```

Comment:
as this array goes to `classnames`, I would change it into an object like:
```
const classes = {
  'inline-help__results-item': true,
  'is-selected': this.props.selectedResult === index,
};
```

or have the first class assigned statically and the object would be just for the selection part:

```
const classes = { 'is-selected': this.props.selectedResult === index };

<li … className={ classNames( 'inline-help__results-item', classes ) }>
```